### PR TITLE
Strings

### PR DESCRIPTION
**Pull request by othane**
pull my strings and I'll go far

make a char array different from a byte array so we can display strings nicely

### CODE REVIEW

**Comment by djpnewton** [comment]
impossible to merge, some of the files have all the line endings changed

**Comment by othane** [comment]
OK i think this should be good to merge now, no more conflicts. Give it a wirl

**Comment by djpnewton** [comment]
line endings in tools/Symbols.cs is borked

some stray tabs mixed with spaces

File: msvc/kowhai_sharp/KowhaiTree.cs [hunk 1]
```diff
@@ -119,6 +121,13 @@ private object GetDataValue(KowhaiNodeInfo info)
                         return (sbyte)data[info.Offset];
                     case Kowhai.UINT8:
                         return data[info.Offset];
+                    case Kowhai.CHAR:
+                        // convert byte array to string
```
Comment (djpnewton) [inline msvc/kowhai_sharp/KowhaiTree.cs:125]: use native function to convert byte array to string

```
   string System.Text.Encoding.UTF8.GetString(byte[])
```

File: msvc/kowhai_sharp/KowhaiTree.cs [hunk 2]
```diff
@@ -136,6 +145,7 @@ private object GetDataValue(KowhaiNodeInfo info)
 
         private byte[] TextToData(string text, ushort dataType)
         {
+            byte[] result = new byte[text.Length + 1];  // +1 for NULL
```
Comment (djpnewton) [inline msvc/kowhai_sharp/KowhaiTree.cs]: this should only apply to the char case

File: src/kowhai.c [hunk 3]
```diff
@@ -21,6 +21,7 @@ int kowhai_get_node_type_size(uint16_t type)
         // normal types to describe a buffer
         case KOW_INT8:
         case KOW_UINT8:
+		case KOW_CHAR:
```
Comment (djpnewton) [inline src/kowhai.c]: tab

File: msvc/kowhai_sharp/KowhaiTree.cs [hunk 4]
```diff
@@ -152,6 +162,12 @@ private byte[] TextToData(string text, ushort dataType)
                     return BitConverter.GetBytes(Convert.ToUInt32(text));
                 case Kowhai.FLOAT:
                     return BitConverter.GetBytes(Convert.ToSingle(text));
+                case Kowhai.CHAR:
```
Comment (djpnewton) [inline msvc/kowhai_sharp/KowhaiTree.cs:164]: should use native string to byte array function

```
        ASCIIEncoding enc = new ASCIIEncoding();
        target = enc.GetString(targetBuf, 0, targetBufferSize);
```

File: src/kowhai.h [hunk 5]
```diff
@@ -23,6 +23,7 @@ enum kowhai_node_type
     KOW_INT32,
     KOW_UINT32,
     KOW_FLOAT,
+	KOW_CHAR,
```
Comment (djpnewton) [inline src/kowhai.h]: tab

**Pull request merged**